beater/beater.go

```diff
@@ -45,11 +45,14 @@ func (bt *ApmServer) Run(b *beat.Beat) error {
 		go bt.client.PublishAll(events)
 	}
 
-	bt.server, err = server.New(bt.config.Server)
-	if err != nil {
+	ucfg := bt.config.ApmServerConfig
+	apmServerConfig := server.DefaultConfig
+	if err := ucfg.Unpack(&apmServerConfig); err != nil {
```

Comment:
Hmm, I don't think so. Based on what is that?

I think there is no more coupling between beater and server than before: 
+ Beater already knew about server (which is ok): https://github.com/elastic/apm-server/blob/master/beater/apm-server.go#L48
+ Server doesn't know about Beater.

Besides, knowledge about the config is *not shared outside. The thing is just unpacked, not inspected. We have the same pattern in other parts of the code base.

Just this snippet actually allows me to remove 2 dependencies:

https://github.com/elastic/apm-server/pull/78/files#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bL18

https://github.com/elastic/apm-server/pull/78/files#diff-110fdf0113b0dcfcdca7c9352c1b4c4fL6

So circular imports are actually less likely, mostly because the beater config now is where it belongs.

`DefaultConfig` is turned public, yes, just like `DefaultBeaterConfig` which was already public before, I don't see how it is different. In turn, now I can make both `DefaultBeaterConfig` and `BeaterConfig` private, so encapsulation game is stronger `2 - 1` :D

Does this make sense or is there anything here that you don't think is true?